File: java/src/main/java/com/lacombe/promo3/taxi/ArrivalHour.java
package com.lacombe.promo3.taxi;

import java.util.Comparator;

public class ArrivalHour implements Comparable<ArrivalHour>{
    private final int hour;

    public ArrivalHour(int hour) {
        this.hour = hour;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        ArrivalHour that = (ArrivalHour) o;

        return hour == that.hour;
    }

    @Override
    public int hashCode() {
        return hour;
    }

    @Override
    public String toString() {
        return Integer.toString(hour);
    }

    @Override
    public int compareTo(ArrivalHour o) {
        return this.hour - o.hour;
    }
}

File: dotnet/Socrates/TaxiOrganization/ArrivalHour.cs
using System;

namespace Socrates.TaxiOrganization
{
    public class ArrivalHour : IComparable<ArrivalHour>
    {
        private readonly int hour;

        public ArrivalHour(int hour)
        {
            this.hour = hour;
        }

        public override string ToString()
        {
            return hour.ToString();
        }

        public override bool Equals(object obj)
        {
            var hour = obj as ArrivalHour;
            return hour != null &&
                   this.hour == hour.hour;
        }

        public override int GetHashCode()
        {
            return -2145515161 + hour.GetHashCode();
        }

       

        public int CompareTo(ArrivalHour other)
        {
            return this.hour.CompareTo(other.hour);
        }
    }
}
